test/CQRS/LeanCode.CQRS.AspNetCore.Tests/ServiceProviderRegistrationExtensionsTests.cs

```diff
@@ -0,0 +1,161 @@
+using LeanCode.CQRS.Validation;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace LeanCode.CQRS.AspNetCore.Tests;
+
+public class ServiceProviderRegistrationExtensionsTests
+{
+    private static readonly TypesCatalog ThisCatalog = TypesCatalog.Of<ServiceProviderRegistrationExtensionsTests>();
+
+    [Fact]
+    public void Registers_base_services()
+    {
+        var sp = BuildProvider();
+
+        sp.Should().HaveService<ICQRSObjectSource>();
+        sp.Should().HaveService<CQRSMetrics>();
+        sp.Should().HaveService<ISerializer>();
+        sp.Should().HaveService<RoleRegistry>();
+        sp.Should().HaveService<IHasPermissions>();
+        sp.Should().HaveService<ICommandValidatorResolver>();
+    }
+
+    [Fact]
+    public void Registers_handlers()
+    {
+        var sp = BuildProvider();
+
+        sp.Should().HaveService<Local.LocalCommandHandler>();
+        sp.Should().HaveService<Local.LocalQueryHandler>();
+        sp.Should().HaveService<Local.RemoteQueryHandler>();
+    }
+
+    [Fact]
+    public void Does_not_register_local_executors_by_default()
+    {
+        var sp = BuildProvider();
+
+        sp.Should().NotHaveService<ILocalCommandExecutor>();
+        sp.Should().NotHaveService<ILocalQueryExecutor>();
+        sp.Should().NotHaveService<ILocalOperationExecutor>();
+    }
+
+    [Fact]
+    public void Registers_local_executors_on_request()
+    {
+        var sp = BuildProvider(
+            c => c.WithLocalCommands(_ => { }).WithLocalQueries(_ => { }).WithLocalOperations(_ => { })
+        );
+
+        sp.Should().HaveService<ILocalCommandExecutor>();
+        sp.Should().HaveService<ILocalQueryExecutor>();
+        sp.Should().HaveService<ILocalOperationExecutor>();
+    }
+
+    [Fact]
+    public void Registers_keyed_local_executors_on_request()
+    {
+        var sp = BuildProvider(
+            c =>
+                c.WithKeyedLocalCommands("commands", _ => { })
+                    .WithKeyedLocalQueries("queries", _ => { })
+                    .WithKeyedLocalOperations("operations", _ => { })
+        );
+
+        sp.Should().HaveKeyedService<ILocalCommandExecutor>("commands");
+        sp.Should().HaveKeyedService<ILocalQueryExecutor>("queries");
+        sp.Should().HaveKeyedService<ILocalOperationExecutor>("operations");
+    }
+
+    [Fact]
+    public void Replaces_serializer_service()
+    {
+        var serializer = new CustomSerializer();
+        var sp = BuildProvider(c => c.WithSerializer(serializer));
+
+        sp.GetRequiredService<ISerializer>().Should().BeSameAs(serializer);
+    }
+
+    private static ServiceProvider BuildProvider(Action<CQRSServicesBuilder>? configure = null)
+    {
+        var collection = new ServiceCollection();
+        var builder = collection.AddCQRS(ThisCatalog, ThisCatalog);
+        configure?.Invoke(builder);
+        return collection.BuildServiceProvider();
+    }
+
+    internal class CustomSerializer : ISerializer
```

Comment:
We'd better do what he asks for - I don't want it to be stacking at every PR.

Reply:
I can't make it report it reliably on my dev machine and find about that on CI only.

Reply:
Fixed